src/pgduckdb_types.cpp

```diff
@@ -317,8 +485,10 @@ ConvertDuckToPostgresArray(TupleTableSlot *slot, duckdb::Value &value, idx_t col
 	auto arr = OP::ConstructArray(datums, nulls, number_of_dimensions, dimensions, lower_bounds);
 
 	// Free allocated memory
-	pfree(datums);
-	pfree(nulls);
+    if (append_state.expected_values > 0) {
+        pfree(datums);
+        pfree(nulls);
+    }
```

Comment:
I don't think this change should be necessary.

Reply:
I also think the `pfree` itself can handle this. But when I remove the if condition, and run the tests. I'll get the `CREATE TABLE char_array_2d(a CHAR(1)[][]);
INSERT INTO char_array_2d VALUES
    ('{{"a","b"},{"c","d"}}'),
    ('{{"e","f","g"},{"h","i","j"}}'),
    (NULL),
    ('{{"k","l"},{"m",NULL}}'),
    ('{}');
SELECT * FROM char_array_2d;
server closed the connection unexpectedly
	This probably means the server terminated abnormally
	before or while processing the request.
connection to server was lost`. That's why I add this.